### PR TITLE
Completed Lesson 09 assignment. Included are the mailroomOO file and …

### PR DESCRIPTION
**Pull request by letoui**
…the associated test file.

### CODE REVIEW

File: students/ian_letourneau/mailroom/mailroomOO.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    def __init__(self, name, donations=None):
+        """Initialization for Donor objects.
+
+        Params name: Will be the name given to Donor object
+        Params donations: a value or list of values to be 
+            combined into donation_list attribute."""
+        self.name = name
+        if isinstance(donations, list):
+            self.donation_list = donations
+        elif donations == None:
```
Comment (natasha-aleksandrova) [inline students/ian_letourneau/mailroom/mailroomOO.py]: it is more pythonic to use `is` to test for None:
`elif donations is None` 

to simplify this:
```
if isinstance(donations, list):
    self.donation_list = donations
else:
    self.donations_list = [donations] if donations else []
```

File: students/ian_letourneau/mailroom/mailroomOO.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    def __lt__(self, other):
+        """Override less than comparison method."""
+        return self.total < other.total
+
+    def send_thank_you(self, donation):
+        """A function to send thank you letter utilizing given amount and
+        referenced donor object."""
+
+        # Append the amount to the running donation_list.
+        self.append(donation)
```
Comment (natasha-aleksandrova) [inline students/ian_letourneau/mailroom/mailroomOO.py]: it is probably best for this function to be `get_thank_you_letter` and keep append logic outside of this as it will feels a little bit like a side-effect.

File: students/ian_letourneau/mailroom/mailroomOO.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        params *args: take in all donor objects and create a list
+            of donors."""
+        self.out_list = []
+        list.__init__(self, list(args))
+
+    def longest_name(self):
+        """Return length od longest donor name in DList."""
+        names = []
+        for donor in self:
+            names.append(donor.name)
```
Comment (natasha-aleksandrova) [inline students/ian_letourneau/mailroom/mailroomOO.py]: could you use a list comprehension here?

File: students/ian_letourneau/mailroom/mailroomOO.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    def send_all(self):
+        """A function that sends a Thank You letter to every donor in the form of
+        creating and writing to a .txt file titled with their name and the current
+        date. The .txt includes their name and their total donations within the
+        thank you memo."""
+        now = datetime.datetime.now()
+        letters = []
+        # Loop through each donor in the list and calculate total donations.
+        for donor in self:
+            donorFile = donor.name.replace(" ", "")
```
Comment (natasha-aleksandrova) [inline students/ian_letourneau/mailroom/mailroomOO.py]: var names should be snake_case -> `donorFile`

File: students/ian_letourneau/mailroom/mailroomOO.py [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+            # Then create file and write thank you letter within open .txt
+            # file.
+            filename = "{}_{}_{}_{}.txt".format(
+                donorFile, now.year, now.month, now.day)
+            f = open(filename, "w+")
+            f.write('Dear {}, we wanted to say thank you for your generous'
+                    ' total donations of ${:.2f}! We hope to continue seeing'
+                    ' you in the future!'.format(
+                        donor, float(donor.total)))
+            f.close()
```
Comment (natasha-aleksandrova) [inline students/ian_letourneau/mailroom/mailroomOO.py]: remember that it is preferred to use `with` statement when working with files.

File: students/ian_letourneau/mailroom/test_mailroomOO.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+# Ian Letourneau
+# 6/8/2018
+# Test script for mailroomOO.py
+
+import mailroomOO as m
```
Comment (natasha-aleksandrova) [inline students/ian_letourneau/mailroom/test_mailroomOO.py]: module names should be all lower case snake_case style -> `mailroom_oo`

File: students/ian_letourneau/mailroom/test_mailroomOO.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def test_donor_creation():
+    assert d1.name == ("LeBron James")
+    assert d1.total == (6609558.5)
+
+# Test send_thank_you donor function
+
+
+def test_send_thank_you():
```
Comment (natasha-aleksandrova) [inline students/ian_letourneau/mailroom/test_mailroomOO.py]: your functions should have 2 lines space between them and if you want a comment for a function, it should go after the function definition:
```
def test_donor_creation():
    """Test send_thank_you donor function"""
    pass
```
but test function names are pretty self explanatory 🙂
